src/openfermion/utils/_trotter_exp_to_qgates.py

```diff
@@ -0,0 +1,277 @@
+    if trotter_order > 3 or trotter_order <= 0:
+        raise ValueError("Invalid trotter order: " + str(trotter_order))
+    if isinstance(k_exp, int):
+        k_exp = float(k_exp)
+    if not isinstance(hamiltonian, QubitOperator):
+        raise TypeError("Hamiltonian must be a QubitOperator.")
+    if not hamiltonian.terms:
+        raise TypeError("Hamiltonian must be a non-empty QubitOperator.")
+    if term_ordering is None:
+        # To have consistent default behavior, ordering = sorted keys.
+        term_ordering = sorted(list(hamiltonian.terms.keys()))
+    if len(term_ordering) == 0:
+        raise TypeError("term_ordering must None or non-empty list.")
+
+    ret_val = []
+    # First order trotter
+    if trotter_order == 1:
+        for step in range(trotter_number):
+            for op in term_ordering:
+                yield QubitOperator(op,
+                                    hamiltonian.terms[op]
+                                    * float(k_exp)
+                                    / float(trotter_number)
+                                    )
+    # Second order trotter
+    elif trotter_order == 2:
+        if len(term_ordering) < 2:
+            raise ValueError("Not enough terms in the Hamiltonian to do " +
+                             "second order trotterization")
+        for step in range(trotter_number):
+            for op in term_ordering[:-1]:
+                yield QubitOperator(op,
+                                    (hamiltonian.terms[op]
+                                    * k_exp
+                                    / (2.0 * float(trotter_number))))
+
+            yield QubitOperator(term_ordering[-1],
+                                hamiltonian.terms[term_ordering[-1]]
+                                * float(k_exp)
+                                / float(trotter_number))
+
+            for op in reversed(term_ordering[:-1]):
+                yield QubitOperator(op, hamiltonian.terms[op]
+                                    * float(k_exp)
+                                    / (2.0 * float(trotter_number)))
+
+    # Third order trotter
+    elif trotter_order == 3:
+        if len(term_ordering) < 2:
+            raise ValueError("Not enough terms in the Hamiltonian to do " +
+                             "third order trotterization")
+        # Make sure original hamiltonian is not modified
+        ham = hamiltonian / float(trotter_number)
+        ham *= float(k_exp)
+        ham_temp = []
+        for term in term_ordering:
+            ham_temp.append(QubitOperator(term, ham.terms[term]))
+        for step in range(trotter_number):
+            for returned_op in _third_order_trotter_helper(ham_temp):
+                yield returned_op
+
+
+
+def pauli_exp_to_qasm(qubit_operator_list,
+                      evolution_time=1.0):
+    """Exponentiate a list of QubitOperators
+    and yield string generators in QASM format
+    using the formula:  exp(evolution_time * op)
```

Comment:
is op assumed to be Hermitian or Anti-Hermitian here?

Reply:
I don't think we're assuming Hermitian, but I am changing it to exp(-1.0j * evolution_time * op)

Reply:
I mean, yes, I suppose it's assumed to be Hermitian.